src/linear_algebra/gram_schmidt_ortho.lean

```diff
@@ -0,0 +1,261 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao, Kevin Buzzard
+-/
+
+import analysis.inner_product_space.projection
+
+/-!
+# Gram-Schmidt Orthogonalization and Orthonormalization
+
+In this file we introduce Gram-Schmidt Orthogonalization and Orthonormalization.
+
+The Gram-Schmidt process takes a set of vectors as input
+and outputs a set of orthogonal vectors which have the same span.
+
+## Main results
+
+- `gram_schmidt_process` : Gram-Schmidt process
+- `gram_schmidt_process_orthogonal` :
+  the proof that "gram_schmidt_process" produces an orthogonal system of vectors
+- `gram_schmidt_process_normed` :
+  Normalized "Gram-Schmidt" (i.e each vector in this system has unit length)
+- `gram_schmidt_process_orthornormal` :
+  the proof that "gram_schmidt_process_normed" produces an orthornormal system of vectors
+- `gram_schmidt_process_span_eq` :
+  the proof that `gram_schmidt_process` preserves the span of vectors
+
+## Notation
```

Comment:
I don't think we actually do this in docstrings for "local" notation

Reply:
oh yes, I changed that, thank you!